lib/rubocop/cop/rails/orm_abstraction.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Avoid breaking ORM abstraction.
+      # @example
+      #
+      #   # bad
+      #   where(foo_id: foo.id)
+      #
+      #   # bad
+      #   create(foo_id: foo.id)
+      #
+      #   # good
+      #   where(foo: foo)
+      #
+      #   # good
+      #   create(foo: foo)
+
+      class ORMAbstraction < Cop
+        def_node_matcher :bad_pair?, <<-PATTERN
+          (pair _ (send !nil? {:id :uuid}))
+        PATTERN
+
+        def on_pair(node)
+          return unless bad_pair?(node)
+          return unless node.key.source.match(BAD_RXP)
+
+          add_offense(
+            node,
+            location: node.loc.expression,
+            message: "prefer `#{prefered_pair(node)}`."
+          )
+        end
+
+        BAD_RXP = /(_id|_uuid)(:|'|"|\s|$)/
```

Comment:
Please, replace the regexp usage with some node manipulation code. This regexp is pretty brittle.

Reply:
I can't find anything about node manipulation. All the cops I see are using string substitutions.

Reply:
I am able to make the regexp more targeted with a better node matcher though.
`          (pair {str_type? sym_type? dstr_type?} (send !nil? {:id :uuid}))`
`        BAD_RXP = /_(id|uuid)(?="|'|$)/`